db/active_replicator_common_collections.go

```diff
@@ -30,26 +31,43 @@ func (arc *activeReplicatorCommon) _initCollections() ([]replicationCheckpoint,
 	}
 
 	if arc.config.CollectionsLocal != nil {
-		for _, scopeAndCollectionName := range arc.config.CollectionsLocal {
-			scopeName, collectionName, err := parseScopeAndCollection(scopeAndCollectionName)
+		for i, localScopeAndCollection := range arc.config.CollectionsLocal {
+			localScopeName, localCollectionName, err := parseScopeAndCollection(localScopeAndCollection)
 			if err != nil {
 				return nil, err
-			} else if scopeName == nil || collectionName == nil {
-				return nil, fmt.Errorf("scope and collection name must be specified: %q - %v", scopeAndCollectionName, arc.config.CollectionsLocal)
+			} else if localScopeName == nil || localCollectionName == nil {
+				return nil, fmt.Errorf("scope and collection name must be specified: %q - %v", localScopeAndCollection, arc.config.CollectionsLocal)
+			}
+			localCollectionsKeyspaces = append(localCollectionsKeyspaces, base.ScopeAndCollectionName{Scope: *localScopeName, Collection: *localCollectionName})
+
+			// remap collection name to remote if set
+			if remoteScopeAndCollection := arc.config.CollectionsRemote[i]; remoteScopeAndCollection != "" {
+				base.DebugfCtx(arc.ctx, base.KeyReplicate, "Mapping local %q to remote %q", localScopeAndCollection, remoteScopeAndCollection)
+				localScopeAndCollection = remoteScopeAndCollection
+
+				remoteScopeName, remoteCollectionName, err := parseScopeAndCollection(remoteScopeAndCollection)
+				if err != nil {
+					return nil, err
+				} else if remoteScopeName == nil || remoteCollectionName == nil {
+					return nil, fmt.Errorf("scope and collection name must be specified: %q - %v", remoteScopeAndCollection, arc.config.CollectionsLocal)
+				}
+				remoteCollectionsKeyspaces = append(remoteCollectionsKeyspaces, base.ScopeAndCollectionName{Scope: *remoteScopeName, Collection: *remoteCollectionName})
+			} else {
+				remoteCollectionsKeyspaces = append(remoteCollectionsKeyspaces, base.ScopeAndCollectionName{Scope: *localScopeName, Collection: *localCollectionName})
 			}
-			getCollectionsKeyspaces = append(getCollectionsKeyspaces, base.ScopeAndCollectionName{Scope: *scopeName, Collection: *collectionName})
 			getCollectionsCheckpointIDs = append(getCollectionsCheckpointIDs, arc.CheckpointID)
 		}
 	} else {
 		// collections to replicate wasn't set - so build a full set based on local database
 		for _, dbCollection := range arc.blipSyncContext.blipContextDb.CollectionByID {
-			getCollectionsKeyspaces = append(getCollectionsKeyspaces, base.ScopeAndCollectionName{Scope: dbCollection.ScopeName, Collection: dbCollection.Name})
+			localCollectionsKeyspaces = append(localCollectionsKeyspaces, base.ScopeAndCollectionName{Scope: dbCollection.ScopeName, Collection: dbCollection.Name})
```

Comment:
I don't think this matters but because this is a map iteration, these collections are going to show up in a different order each time. For practical purposes I don't see this mattering, but for debugging purposes, maybe?

Reply:
Fair comment - like you say, it doesn't really matter what order both sides agree to, as long as it's the same, but I'll log at debug what slice order we do attempt.